EIPS/eip-4788.md

```diff
@@ -29,7 +29,8 @@ restaking constructions, smart contract bridges, MEV mitigations and more.
 |---                           |---                                           |---
 | `FORK_TIMESTAMP`             | TBD                                          |
 | `HISTORY_STORAGE_ADDRESS`    | `Bytes20(0xB)`                               |
-| `G_beacon_root`              | 2100                                         | gas
+| `G_beacon_root`              | 4200                                         | gas
+| `HISTORICAL_ROOTS_LENGTH`    | 98304                                        |
```

Comment:
I think this name is now a bit misleading:

```suggestion
| `HISTORICAL_ROOTS_LENGTH`    | 8192                                        |
| `HISTORICAL_ROOTS_MODULO`    | `HISTORICAL_ROOTS_LENGTH * 12` (98304)                                        |
```

(Then change elsewhere in EIP as well).

Otherwise people think 98304 roots are saved, which is not the case. (Also note that this "only" works if blocks have a constant drift time of `12 * x` (`x` integer) - what happens if we at some point change this drift time..? If we change it to 6 seconds we save twice as much. Could also note that this is thus the max roots ever saved on the chain, which is nice since it upper bounds it)

Reply:
I'd rather just call the 98304 number the `HISTORICAL_ROOTS_MODULUS`

there is a bit of a hidden dependency on the 12 number which comes from the `SECONDS_PER_SLOT` constant on the consensus layer but part of the design conversation around this EIP wants to avoid mixing the CL constant into the EL

we could change this slot time and it would have implications for this EIP in terms of the number of roots stored but (1) this does set an upper bound and (2) I don't think we will change the slot time so dramatically that it drastically changes the effective window a user of this precompile has to use

worst case we can always adjust how this precompile works if we do something like adjusting the slot time